### PR TITLE
feat(celery): add support for Python 3.11, 3.12

### PR DESCRIPTION
**Pull request by Yun-Kim**
Officially adds support for Python 3.11 and 3.12 for the Celery integration.

~~Note: looks like an issue with task exceptions that didn't translate well to Python 3.11. We'll have to investigate this first.~~
- ~~It looks like `celery>=5.3` uses a newer version of `billiard>=4.1.0` and also changed how `billiard.ExceptionInfo` gets constructed in [celery's `handle_failures` callback](https://github.com/celery/celery/blame/main/celery/app/trace.py#L215-L228). Note we were previously using `billiard==3.6.4` and `celery==5.2.7`.~~
- ~~This is now changing the value of the exception `einfo` we check for in our traced [`trace_failure` callback](https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/contrib/celery/signals.py#L189-L193), which makes things weird when we try to ignore some Exceptions in our integration tests for celery.~~

~~I've verified this hypothesis (that the new release of `celery>=5.3.0` changed how they construct `ExceptionInfo` in `handle_failures()` breaks our integration tests) by updating the Python 3.10 test suite to use `celery==5.3.1`, and this also results in the same failures. We'll need to make a bigger effort to make our integration compatible with the latest `celery>=5.3.0`.~~

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by mabdinur — COMMENTED** [review]
@Yun-Kim, is there a feature request for supporting celery for python3.11? Do we need to prioritize this work?

**Comment by Yun-Kim** [comment]
@mabdinur we got an indirect FR in this issue: https://github.com/DataDog/dd-trace-py/issues/4149#issuecomment-1620150435

**Comment by kochb** [comment]
Should I understand that there is a formal feature request procedure that would help with resuming work on this? We're a datadog customer using Celery under Python 3.11, and we're currently encountering the `‘_Code’ object has no attribute ‘co_positions’` error mentioned in #4149.

**Comment by Yun-Kim** [comment]
Hi @kochb, thanks for reaching out! 

> Should I understand that there is a formal feature request procedure that would help with resuming work on this?

Yes, making a feature request via a support ticket would be the fastest way to resume work on this. Feel free to create an issue on this repository as well!

File: riotfile.py [hunk 1]
```diff
@@ -559,10 +559,7 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
                     },
                 ),
                 Venv(
-                    # celery added support for Python 3.10 in 5.2, no official support for 3.11 yet
-                    # Billiard dependency is incompatible with Python 3.11
-                    # https://github.com/celery/billiard/issues/377
-                    pys="3.10",
+                    pys=select_pys(min_version="3.10"),
```
Comment (mabdinur) [inline riotfile.py:562]:
```suggestion
                    pys=["3.10",  "3.11"),
```

**Review by emmettbutler — APPROVED** [review]

**Review by Yun-Kim — COMMENTED** [review]
Looks like tests are passing for 3.11 and 3.12. Thanks @mabdinur for taking this over to the finish line! 🙇

File: releasenotes/notes/support-3-11-celery-39053a6ccb7de942.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    celery: Adds support for Python 3.11 and 3.12
```
Comment (Yun-Kim) [inline releasenotes/notes/support-3-11-celery-39053a6ccb7de942.yaml:4]:
```suggestion
    celery: Adds Python 3.11 and 3.12 support for the celery integration.
```

**Review by emmettbutler — APPROVED** [review]

**Review by mabdinur — APPROVED** [review]

**Comment by kochb** [comment]
Thank you for your work on this!

**Pull request merged**